exchange/auction_test.go

```diff
@@ -280,6 +280,227 @@ func runCacheSpec(t *testing.T, fileDisplayName string, specData *cacheSpec) {
 	}
 }
 
+type testNewAuctionSpec struct {
```

Comment:
This type has no use outside of the `TestNewAuction` method. Why not use an anonymous struct?
```
testCases := []struct {
	description     string
	seatBids        map[openrtb_ext.BidderName]*pbsOrtbSeatBid
	numImps         int
	preferDeals     bool
	expectedAuction auction,
}{
	{
	...
	},
}
```